docs/src/contribute/design/next-tedge-api/topic-examples.md

```diff
@@ -0,0 +1,856 @@
+---
+title: Topic Examples
+tags: [Documentation]
+sidebar_position: 2
+---
+
+# Examples
+
+The following examples will demonstrate how the flexible topic structure can be used to provide rich modelling tools to publish data which is more observable to other components.
+
+## Example: Default device/service topic semantics
+
+### Publish to the main device
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main///m/environment' '{
+  "temperature": 23.4
+}'
+```
+
+If the there is no measurement type, then the type can be left empty, but it must have the trailing slash `/` (so that the number of topic segments is the same).
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main///m/' '{
+  "temperature": 23.4
+}'
+```
+
+### Publish to a child device
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/child01///m/environment' '{
+  "temperature": 23.4
+}'
+```
+
+### Publish to a service on the main device
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main/service/nodered/m/environment' '{
+  "temperature": 23.4
+}'
+```
+
+Any MQTT client can subscribe to all measurements for all entities (devices and services) using the following MQTT topic:
+
+```sh te2mqtt
+tedge mqtt sub 'te/+/+/+/+/m/+'
+```
+
+If you want to be more specific and only subscribe to the main device, then you can used fixed topic names rather than wildcards:
+
+```sh te2mqtt
+tedge mqtt sub 'te/device/main/+/+/m/+'
+```
+
+Or to subscribe to a specific type of measurement published to an services on the main device, then use:
+
+```sh te2mqtt
+tedge mqtt sub 'te/device/main/service/+/m/memory'
+```
+
+### Publish to a service on a child device
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/child01/service/nodered/m/environment' '{
+  "temperature": 23.4
+}'
+```
+
+### Check which entities have been registered
+
+A list of registered devices can be retrieved by using a simple subscription:
+
+```sh te2mqtt
+tedge mqtt sub 'te/+/+'
+```
+
+Or if you want to listen to all devices and services, then use the following:
+
+```sh te2mqtt
+tedge mqtt sub 'te/+/+/+/+'
+```
```

Comment:
There is something implied here: that registration data should not be published neither on `'te/+` nor `'te/+/+/+`.

I'm okay with that but in that case we need to be more explicit documenting the entity/component topics.

Reply:
After the review, I'll update the example to include subscribing to all 4 levels of the identifier group, e.g.

```sh
mosquitto_sub -t 'te/+' -t 'te/+/+' -t 'te/+/+/+' -t 'te/+/+/+/+'
```